Mg dev

LindsellR opened the pull request:
I've added the login-view, sign-up view folders, and .jsx files in the src/components file, with form validation, authorisation added and tokens stored in localStorage. Both are integrated to the main-view.jsx. After sign-up, then login, users see the displayed list of films and the logout button. Click on the titles displays the films details and a back button.

LindsellR committed:
main-view.jsx, sign-up, login

created similarMovies component

added sign-up and login.jsx pages

LindsellR committed:
Signup-view, login-view, main-view.jsx

Added login-view.jsx and signup-view.jsx, applied form validation and authorisation with login details stored in localStorage. Integrated both with main-view.jsx

Glenzy commented on src/components/login-view/login-view.jsx:
```diff
@@ -0,0 +1,61 @@
+          type="password"
+          value={password}
+          onChange={(e) => setPassword(e.target.value)}
+          minLength="5"
```
This should not be here. You also aren’t enforcing this in the `<SignupView />` component. This causes an issue because your API doesn’t enforce these validations. I signed up with a 4 letter password and so was blocked from logging in.

Glenzy commented on src/components/main-view/main-view.jsx:
```diff
@@ -1,37 +1,118 @@
+      .then((response) => {
+        if (!response.ok) {
+          return response.text().then((text) => {
+            throw new Error(`Error ${response.status}: ${text}`);
```
This is dangerous without properly handling this error. Once you throw there’s no where to go unless you catch it, which you’re not. Try returning a standardised error response that you can save and display to the user instead. Something like:

```
const [error, setError] = useState(undefined);

// in useEffect
 fetch(
  if (!response.ok) {
    setError(’No movies today, please try again’);
    return;
  }
).then((movieData)=>{
  if(! movieData) return;
  setError(undefined);
})
```

then somewhere in your UI you could have a nice error handling without destroying your app. .

Glenzy commented on src/components/signup-view/signup-view.jsx:
```diff
@@ -0,0 +1,76 @@
+          value={username}
+          onChange={(e) => setUsername(e.target.value)}
+          required
+          minLength="5"
```
Again this one is not enforced in the API and it’s also bad practice to enforce a min length on names in general. This should probably be removed so your API can handle any incorrect input.

LindsellR committed:
Login-view, signup-view, main-view.jsx

Removed min length for Usename in sign-up/login.

Fixed error handling in main-view and error messages.

Glenzy reviewed (approved):
Looks good!
